tests/test_proc/test_aggregate.py

```
@@ -263,3 +263,112 @@ def test_single_reduction(self, random_patch):
+        out = patch.idxmax("time", dim_reduce="squeeze")
+        assert np.array_equal(np.asarray(out.data), np.zeros(2))
+
+    @pytest.mark.parametrize("dim", [None, ["time"], ("time", "distance")])
+    def test_non_string_dim_raises(self, random_patch, dim):
+        """These reduce exactly one dimension, named by a string."""
+        with pytest.raises(ParameterError, match="single dimension"):
+            random_patch.idxmax(dim)
+
+    def test_bad_dim_raises(self, random_patch):
+        """A dimension the patch does not have is an error."""
+        with pytest.raises(Exception):
```

Ruff (0.16.2)

[warning] Do not assert blind exception: `Exception`

(B017)